### PR TITLE
add method to set settings

### PR DESCRIPTION
**Pull request by andrej-sajenko**
add method to add elasticsearch settings

### CODE REVIEW

File: georocket-server/src/main/java/io/georocket/index/elasticsearch/ElasticsearchClient.java [hunk 1]
```diff
@@ -129,7 +129,15 @@
...
   Observable<Void> ensureIndex();
-  
+
+  /**
+   * Add Elasticsearch index settings.
+   * @param settings the settings to set for an index.
+   * @return on observable emitting <code>true</code> if the operation
+   * was acknowledged by Elasticsearch, <code>false</code> otherwise.
+   */
+  Observable<Boolean> putSettings(JsonObject settings);
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/index/elasticsearch/ElasticsearchClient.java]: In the Elasticsearch docs it is called 'update settings' and not 'put settings'

File: georocket-server/src/main/java/io/georocket/index/elasticsearch/RemoteElasticsearchClient.java [hunk 2]
```diff
@@ -191,7 +191,35 @@ public void close() {
...
+
+  @Override
+  public Observable<Boolean> putSettings(JsonObject settings) {
+    String indexUri = "/" + index;
+    String settingsUri = indexUri + "/_settings";
+    String openUri = indexUri + "/_open";
+    String closeUri = indexUri + "/_close";
+
+    // Elasticsearch need to close a index befor it can be updated.
+    return performRequestRetry(HttpMethod.POST, closeUri, null)
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/index/elasticsearch/RemoteElasticsearchClient.java]: You only need to close the index if you want to add analyzers. Updating the settings without analyzers is a valid operation that should not require the index to be closed and opened again.

File: georocket-server/src/test/java/io/georocket/index/elasticsearch/RemoteElasticsearchClientTest.java [hunk 3]
```diff
@@ -65,6 +66,20 @@
...
+      "    \"analyzer\":{\n" +
+      "      \"content\":{\n" +
+      "        \"type\":\"custom\",\n" +
+      "        \"tokenizer\":\"whitespace\"\n" +
+      "      }\n" +
+      "    }\n" +
+      "  }\n" +
+      "}");
+
+  private static final JsonObject ACKNOWLEDGED = new JsonObject()
```
Comment (michel-kraemer) [inline georocket-server/src/test/java/io/georocket/index/elasticsearch/RemoteElasticsearchClientTest.java]: Please use this field in all cases where an acknowledgment is returned in this class.

**Comment by andrej-sajenko** [comment]
#81

**Pull request closed without merging**